src/Orleans/Properties/AssemblyInfo.cs

```diff
@@ -49,6 +49,7 @@ MIT License
 [assembly: InternalsVisibleTo("OrleansHost, PublicKey=0024000004800000940000000602000000240000525341310004000001000100eb81539dc2728946d4860bad0fbc91ea23d7a835e00547aa4e9a7abf185535b564d29dbe77915b779951fef2f4b1ab39cc3a246cf572c7b217b5c892835a687eb22638183d66930039ff9fa6e7e71cb880f854fb76b884dfc905bf1f3c623abad7d4732c1ce62efff937155e352170831239acbe103a41ed363a90001d3108ca")]
 [assembly: InternalsVisibleTo("OrleansAzureUtils, PublicKey=0024000004800000940000000602000000240000525341310004000001000100eb81539dc2728946d4860bad0fbc91ea23d7a835e00547aa4e9a7abf185535b564d29dbe77915b779951fef2f4b1ab39cc3a246cf572c7b217b5c892835a687eb22638183d66930039ff9fa6e7e71cb880f854fb76b884dfc905bf1f3c623abad7d4732c1ce62efff937155e352170831239acbe103a41ed363a90001d3108ca")]
 [assembly: InternalsVisibleTo("OrleansManager, PublicKey=0024000004800000940000000602000000240000525341310004000001000100eb81539dc2728946d4860bad0fbc91ea23d7a835e00547aa4e9a7abf185535b564d29dbe77915b779951fef2f4b1ab39cc3a246cf572c7b217b5c892835a687eb22638183d66930039ff9fa6e7e71cb880f854fb76b884dfc905bf1f3c623abad7d4732c1ce62efff937155e352170831239acbe103a41ed363a90001d3108ca")]
+[assembly: InternalsVisibleTo("OrleansProviders, PublicKey=0024000004800000940000000602000000240000525341310004000001000100eb81539dc2728946d4860bad0fbc91ea23d7a835e00547aa4e9a7abf185535b564d29dbe77915b779951fef2f4b1ab39cc3a246cf572c7b217b5c892835a687eb22638183d66930039ff9fa6e7e71cb880f854fb76b884dfc905bf1f3c623abad7d4732c1ce62efff937155e352170831239acbe103a41ed363a90001d3108ca")]
```

Comment:
We have been trying to keep it so that the shipped providers are all user-level code and don't require any privileged internal access to Orleans runtime interfaces / functions.

Could you elaborate on which functionality / APIs were required by the DI provider(s) and not available through public APIs?

Rather than granting privilege elevation to `OrleansProviders.dll` [which would not be available to others writing their own providers] we should instead look at whether /  how to expose the required functionality through the public `IProviderRuntime` interface.